### PR TITLE
luarocks

### PR DESCRIPTION
**Pull request by jerowe**
luarocks recipe.

### CODE REVIEW

File: recipes/luarocks/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: http://luarocks.github.io/luarocks/releases/luarocks-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True # [win]
```
Comment (jakirkham) [inline recipes/luarocks/meta.yaml]: I think the linter wants one more space here before `#`.

**Comment by jakirkham** [comment]
@alexbw , could you please take a look at this? 😄

**Comment by jerowe** [comment]
@alexbw, I completely stole this from your repo. ;-) I'm going to steal a few more, mostly the filesystem and posix.

File: recipes/luarocks/meta.yaml (hunk 1 above)
Comment (jerowe) [inline recipes/luarocks/meta.yaml]: Noted. ;-)

File: recipes/luarocks/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  # Some package might need a `test/commands` key to check CLI.
+  # List all the packages/modules that `run_test.py` imports.
+  commands:
+    - luarocks --lr-path
+
+about:
+  home: https://luarocks.org/
+  license: MIT
+  license_family: MIT
+  # license_file: LICENSE.txt
```
Comment (jakirkham) [inline recipes/luarocks/meta.yaml]: Please replace `LICENSE.txt` with `COPYING`.

File: recipes/luarocks/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+# Jinja variables help maintain the recipe as you'll update the version only here.
+{% set name = "lua-luarocks" %}
+{% set version = "2.4.2" %}
+{% set sha256 = "0e1ec34583e1b265e0fbafb64c8bd348705ad403fe85967fd05d3a659f74d2e5" %}
+
+# sha256 is the prefered checksum -- you can get it for a file with:
+#  `openssl sha256 <file name>`.
+# You may need the openssl package, available on conda-forge
+#  `conda install openssl -c conda-forge``
```
Comment (jakirkham) [inline recipes/luarocks/meta.yaml]: Please cut these comments from the `example`.

File: recipes/luarocks/build.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+# TODO
+# - Would be good if luarocks would offer the ability to use only a user rockspec, instead of system.
+#   (Using system-only rockspec with --force-config flag installs packages with restrictive privileges)
+./configure --prefix=$PREFIX \
+    --sysconfdir=$PREFIX/share/lua/ \
+    --with-lua-include=$PREFIX/include \
+    --rocks-tree=$PREFIX
+
+make bootstrap
```
Comment (jakirkham) [inline recipes/luarocks/build.sh]: Is there a `make check`? Also do we need `make install`?

Comment (jakirkham) [inline recipes/luarocks/build.sh]: Also [this]( https://github.com/luarocks/luarocks/blob/v2.4.2/Makefile#L18-L21 ) seems relevant.

**Comment by jerowe** [comment]
@jakirkham , I have questions for you if you care to answer. ;-)

I'm over on conda/conda-env a lot, and I am always seeing multiuser environments pop up. Technically they work (or they did before conda [4.2.9](https://github.com/conda/conda/issues/3906)), but I'm always wondering why the conda world doesn't push using modules more? Everytime I see this question I pop in with "modules! Unless you are on windows. Then you are screwed".

The only answer I can come up with is that they might be difficult to get working on windows, but after some further research I'm not sure this is the case. 

[Lmod](https://www.tacc.utexas.edu/research-development/tacc-projects/lmod) is a project based on environment modules. It requires lua and tcl. Both of these technologies are supported on windows, but not quite through conda. In the near future I am going to be building a lmod package for conda-forge, as well as an [easybuild](http://hpcugent.github.io/easybuild/) package. But I wonder, why wouldn't these work for windows users?

File: recipes/luarocks/build.sh (hunk 4 above)
Comment (jerowe) [inline recipes/luarocks/build.sh]: I completely stole this from @alexbw and didn't investigate beyond this. I will go and look at the luarocks docs now.

File: recipes/luarocks/meta.yaml (hunk 3 above)
Comment (jerowe) [inline recipes/luarocks/meta.yaml]: done.

File: recipes/luarocks/build.sh (hunk 4 above)
Comment (jerowe) [inline recipes/luarocks/build.sh]: What do you think the functional difference is between a usual install and installing as a rock?

Comment (jerowe) [inline recipes/luarocks/build.sh]: I don't see a make check.

Comment (jerowe) [inline recipes/luarocks/build.sh]: [stack overflow](http://stackoverflow.com/questions/29334171/should-icompile-luarocks-as-usual-or-as-a-rock) tells me that `make bootstrap` installs luarocks as a rock, and can then be upgraded with `luarocks install luarocks`. 

So it seems that we should stick with `make build; make install` in order to stick with the version supplied by conda.

**Comment by scopatz** [comment]
Hi @jerowe, can you please enable CircleCI on your fork so that linux runs: http://conda-forge.github.io/docs/ciservices.html#circleci-linux

Sorry I can't be more help on your modules questions.

**Comment by jerowe** [comment]
@scopatz, sure, but why do I need to enable it? I think it used to just run. Shouldn't it be invoked on the conda-forge side?

**Comment by scopatz** [comment]
@jerowe CircleCI "fixed" a "bug" and they no longer build PRs automatically. It is a huge pain.

**Comment by jerowe** [comment]
The conda forge staged recipes says its building, and I have the option to follow builds.

I haven't used CircleCi and I'm not sure what I am supposed to be doing.

**Comment by scopatz** [comment]
@jakirkham figured out a way around this, but it involves closing an reopening the PR.  Hopefully this works.

**Comment by scopatz** [comment]
Hmm that didn't seem to work.  Can you try following the build, @jerowe?

**Comment by jakirkham** [comment]
Strange response on the webhook. Will try closing and reopening again.

**Comment by jakirkham** [comment]
Hmm...do you have CircleCI enabled on your fork of staged-recipes @jerowe ? It looks like it may be, but am not sure. If so, could you try disabling it?

**Comment by jerowe** [comment]
@jakirkham , sorry for the late response. I got distracted. I don't think I have CircleCI enabled.

**Comment by jerowe** [comment]
@jakirkham , after a closer look I realized I was in fact building on my circle ci. I'll push a commit and see if that triggers a build.

File: recipes/luarocks/build.sh (hunk 4 above)
Comment (jakirkham) [inline recipes/luarocks/build.sh]: Hmm...interesting. I think Alex had added some support to `conda-build` for Lua. Not sure if having this be a rock would be important for that or not. Suppose we could leave it and see what happens.

File: recipes/luarocks/build.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+set -x -e
+
+# TODO
+# - Would be good if luarocks would offer the ability to use only a user rockspec, instead of system.
+#   (Using system-only rockspec with --force-config flag installs packages with restrictive privileges)
```
Comment (jakirkham) [inline recipes/luarocks/build.sh]: Do you know anything about this comment? Not sure if it was Alex's or yours.

**Comment by jakirkham** [comment]
So I don't really know much about what is happening with Lua and unfortunately don't have time to dig in myself. That being said, I'd be more than happy to just merge this and see how it goes. What do you think?

**Comment by jerowe** [comment]
@jakirkham , I don't know how I didn't see this!

Yes, please merge!

File: recipes/luarocks/build.sh (hunk 5 above)
Comment (jerowe) [inline recipes/luarocks/build.sh]: It was alex's.

**Comment by jerowe** [comment]
Hi, @jakirkham. How is it going? How are we doing on merging this?

**Comment by scopatz** [comment]
Sorry about the delay. I figure you can fix things up in the feedstock

File: recipes/luarocks/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
+{% set name = "lua-luarocks" %}
```
Comment (jakirkham) [inline recipes/luarocks/meta.yaml]: Sorry I missed this before. The package name needs to match the directory name where the recipe is stored. Should it be `luarocks` or `lua-luarocks`? Please send a follow-up PR to correct this and ping core.

**Comment by jakirkham** [comment]
> How is it going? How are we doing on merging this? Is there anything else I should do?

Sorry have been out the last week. Also will likely be somewhat unavailable over the next couple weeks. So am sorry if I have slow response times as a result.

**Pull request merged**